daml-assistant/exe/DA/Daml/Assistant.hs

```diff
@@ -287,12 +288,32 @@ runCommand env@Env{..} = \case
             dispatch env path args
 
     Dispatch SdkCommandInfo{..} cmdArgs -> do
-        wrapErr ("Running " <> unwrapSdkCommandName sdkCommandName <> " command.") $ do
+        wrapErrWithEnvVarWarning env ("Running " <> unwrapSdkCommandName sdkCommandName <> " command.") $ do
             sdkPath <- required "Could not determine SDK path." envSdkPath
             let path = unwrapSdkPath sdkPath </> unwrapSdkCommandPath sdkCommandPath
                 args = unwrapSdkCommandArgs sdkCommandArgs ++ unwrapUserCommandArgs cmdArgs
             dispatch env path args
 
+envVarAddedVersion :: ReleaseVersion
+envVarAddedVersion = either throw id $ unsafeParseOldReleaseVersion "2.9.0-snapshot.20240210.0"
+
+wrapErrWithEnvVarWarning :: Env -> Text -> IO a -> IO a
+wrapErrWithEnvVarWarning Env{..} ctx act = do
+    case (envSdkVersion, envProjectPath) of
+        (Just ver, Just path) -> do
+            hasEnvVars <- projectConfigExistsWithEnv path
+            let sdkMissingEnvSupport = ver < envVarAddedVersion && not (isHeadVersion ver)
+            if hasEnvVars && sdkMissingEnvSupport
+                then do
+                    putStrLn "WARNING: Using an SDK that does not support Environment Variable Interpolation with a daml.yaml containing Environment Variables."
```

Comment:
We can decapitalize Environment Variables. It might be good to also include the version of the SDK and the version bound for an SDK that *will* interpolate correctly, so that the user has a path to fix things

Reply:
Yeah I wanna talk about this.

Reply:
I aint gonna do it, you can't make me, in fact, I'm deleting this code
(per dive discussion)

Reply:
I aint gonna do it, you can't make me, in fact, I'm deleting this code
(per dive discussion)
(I will include the sdk version string though)